discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -0,0 +1,324 @@
+import concurrent.futures
+import datetime
+import logging
+import re
+import time
+from typing import Tuple, Optional
+from sqlalchemy import desc, and_
+from solana.publickey import PublicKey
+from src.tasks.celery_app import celery
+from src.utils.config import shared_config
+from src.utils.solana import get_address_pair
+from src.models import User, UserBankTransaction, UserBankAccount
+from src.queries.get_balances import enqueue_balance_refresh
+
+logger = logging.getLogger(__name__)
+
+# Populate values used in UserBank indexing from config
+USER_BANK_ADDRESS = shared_config["solana"]["user_bank_program_address"]
+WAUDIO_PROGRAM_ADDRESS = shared_config["solana"]["waudio_program_address"]
+USER_BANK_KEY = PublicKey(USER_BANK_ADDRESS) if USER_BANK_ADDRESS else None
+WAUDIO_PROGRAM_PUBKEY = (
+    PublicKey(WAUDIO_PROGRAM_ADDRESS) if WAUDIO_PROGRAM_ADDRESS else None
+)
+
+# Static SPL Token Program ID
+SPL_TOKEN_ID = PublicKey("TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA")
```

Comment:
this is actually static, so i don't think it will ever change - but it will likely get referenced throughout so we should maybe put it in a constants file?

Reply:
Move to solana.py file as const